docs/design/resource-states.md

```diff
@@ -0,0 +1,222 @@
+
+   // The reason for the condition's last transition.
+   // Reasons should be CamelCase with no spaces. A reason is always provided, this field will not be empty
+   // +required
+   Reason string `json:"reason"`
+
+   // A human readable message indicating details about the transition. This field may be empty
+   // +optional
+   Message string `json:"message,omitempty"`
+}
+```
+
+### The `Ready` condition
+The above discusses support for arbitrary conditions. In practice at least for now, ASO will only expose a single `Ready` condition.
+The addition of a `Severity` field inspired by [Cluster API](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20200506-conditions.md) allows different combinations of `Severity` and `Reason` to combine together and describe complex scenarios.
+
+#### The `Reason` field
+Inside of the `Ready` condition we always include a `Reason` which provides a programatically consumable reason that the resource is in the given state. 
+`Reason` is derived from multiple sources depending on where in the resource lifecycle we are. The value in `Reason` may be set by the operator itself, or be set to the result of an error or response code received from Azure. 
+
+The operator defines the following reasons:
+
+| Reason                 | Severity                        |   Meaning                                                                                                                                                                 |
+|------------------------|---------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Reconciling            | Info                            | A request has been submitted to Azure. No error has been received from Azure, but one may occur in the future.                                                            |
+| WaitingForOwner        | Warning                         | The `owner` of this resource cannot be found in Kubernetes. Progress is blocked until the `owner` is created.                                                             |
+| Deleting               | Info                            | The resource is being deleted.                                                                                                                                            |
+| Succeeded              | None (""/empty)                 | The `spec` has successfully been applied. No additional changes are being attempted at this time.                                                                         |
+
+There are no failure conditions (`Severity = Error`) specified here because currently all fatal errors come directly from Azure. When an error response is received from Azure, the `Code` from Azure is set as the `Reason`, and the `Message` from Azure
+is set as the `Message`.
```

Comment:
GitHub respects line breaks in markdown, so your paragraphs are going look a little weird when rendered. Each paragraph should be a single line of text. #nitpick

Reply:
Fixed